packages/flutter_tools/lib/src/commands/build.dart

```diff
@@ -198,8 +198,14 @@ class BuildCommand extends FlutterCommand {
       archive.addFile(_createSnapshotFile(snapshotPath));
     }
 
-    for (_Asset asset in assets)
-      archive.addFile(_createFile(asset.key, asset.base));
+    for (_Asset asset in assets) {
+      ArchiveFile file = _createFile(asset.key, asset.base);
+      if (file == null) {
+        stderr.writeln('Cannot find asset "${asset.key}" in directory "${path.absolute(asset.base)}".');
```

Comment:
@devoncarew 's recommendation is not to use the logging system for messages intended for humans.  I don't fully understand when we're supposed to use logging and when we're supposed to just write to stdout/stderr.